src/LinkCollection.php

```diff
@@ -44,7 +44,7 @@ public function getLinksByRel($rel): array
     /**
      * {@inheritDoc}
      */
-    public function withLink(LinkInterface $link): self
+    public function withLink(LinkInterface $link): static
```

Comment:
This is a BC break for downstream consumers that implement this trait, and aren't `final`